eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Utxow.hs

```diff
@@ -333,3 +367,115 @@ shelleyToConwayUtxowPredFailure = \case
   Shelley.ConflictingMetadataHash mm -> ConflictingMetadataHash mm
   Shelley.InvalidMetadata -> InvalidMetadata
   Shelley.ExtraneousScriptWitnessesUTXOW xs -> ExtraneousScriptWitnessesUTXOW xs
+
+conwayPPViewHashesMatch ::
+  forall era.
+  AlonzoEraTx era =>
+  Tx era ->
+  PParams era ->
+  ScriptsProvided era ->
+  Set ScriptHash ->
+  Test (ConwayUtxowPredFailure era)
+conwayPPViewHashesMatch tx pp (ScriptsProvided scriptsProvided) scriptsNeeded = do
```

Comment:
Instead of making a copy of the whole of utxo rule, it will be better if you change implementation of `ppViewHashesMatch` in Alonzo era (also rename it to `checkScriptIntegrityHash`). It will require addition of `ScriptIntegrityHashMismatch` predicate failure in Alonzo and Babbage as well, but it will allow us to completely remove `PPViewHashesDontMatch` once we do the next intra-era hard fork.

If we don't do that we will have to drag with us this dumb `PPViewHashesDontMatch` forever.